topics/go/algorithms/strings/reverse/reverse.go

```diff
@@ -0,0 +1,39 @@
+package main
+
+import (
+	"fmt"
+)
+
+func main() {
+
+	fmt.Println(ReverseString("Hello World"))
+}
+
+func ReverseString(str string) string {
+
+	// Here we create a slice of codePoints.
+	codePoints := []rune(str)
+
+	// Here we create int that will be a pointer to the front of the codePoints.
+	var fontCodePoint int
+
+	// Here  we create int that will be a pointer to the end of the codePoints.
+	backCodePoint := len(codePoints) - 1
+
+	// While fontCodePoint is less than backCodePoint.
+	for fontCodePoint < backCodePoint {
+
+		// Here we swap the values of the slice at the codePoints[fontCodePoint] and codePoints[backCodePoint].
+		codePoints[fontCodePoint], codePoints[backCodePoint] = codePoints[backCodePoint], codePoints[fontCodePoint]
+
+		// Here we increase the value of fontCodePoint by 1.
+		fontCodePoint = fontCodePoint + 1
+
+		// Here we decrease the value of backCodePoint by 1.
+		backCodePoint = backCodePoint - 1
+	}
+
+	// Here we return a new string in reverse order.
+	return string(codePoints)
```

Comment:
Don’t kill me. Put yourself in the shoes of others. Do the readability review. 

Also, we need a test for this.

Reply:
I have added a test.

Reply:
You are super close

Put the three lines together for the swap. Provide one sentence comment about the swap. I think the swap is self evident now.

Put the two lines that increment / decrement beg/end together. Don’t let comments hurt code readability. Those lines are better together. Have one simple comment line for teaching.

Reply:
Completed.